geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommand.java

```diff
@@ -0,0 +1,209 @@
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public ResultModel execute(
+      @CliOption(key = METHOD_AUTHORIZER_NAME,
+          help = METHOD_AUTHORIZER_NAME_HELP) String methodAuthorizerName,
+      @CliOption(key = AUTHORIZER_PARAMETERS,
+          help = AUTHORIZER_PARAMETERS_HELP) String[] authorizerParameters,
+      @CliOption(key = SINGLE_AUTHORIZER_PARAMETER,
+          help = SINGLE_AUTHORIZER_PARAMETER_HELP) String singleAuthorizerParameter) {
+    ResultModel result;
+
+    QueryConfigService queryConfigService = getQueryConfigService();
+    Set<String> parametersSet = null;
+    if (methodAuthorizerName != null) {
+      if (!isSecurityEnabled()) {
+        return ResultModel.createError(SECURITY_NOT_ENABLED_MESSAGE);
+      }
+      if (Boolean.parseBoolean(
+          System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation"))) {
+        return ResultModel.createError(DEPRECATED_PROPERTY_ERROR);
+      }
+      if (authorizerParameters != null) {
+        parametersSet = new HashSet<>(Arrays.asList(authorizerParameters));
+      } else if (singleAuthorizerParameter != null) {
+        parametersSet = new HashSet<>(Collections.singleton(singleAuthorizerParameter));
+      }
+      populateMethodAuthorizer(methodAuthorizerName, parametersSet, queryConfigService);
+    }
+
+    Set<DistributedMember> targetMembers = findMembers(null, null);
+    if (targetMembers.size() > 0) {
+      Object[] args = new Object[] {methodAuthorizerName, parametersSet};
+      List<CliFunctionResult> functionResults =
+          executeAndGetFunctionResult(new AlterQueryServiceFunction(), args,
+              targetMembers);
+      String footer = null;
+      for (CliFunctionResult cliResult : functionResults) {
+        if (!cliResult.isSuccessful()) {
+          footer = PARTIAL_FAILURE_MESSAGE;
+          break;
+        }
+      }
+      result =
+          ResultModel.createMemberStatusResult(functionResults, null, footer, false, true);
+    } else {
+      result = ResultModel.createInfo(NO_MEMBERS_FOUND_MESSAGE);
+    }
+    result.setConfigObject(queryConfigService);
+    return result;
+  }
+
+  boolean isSecurityEnabled() {
+    return ((InternalCache) CacheFactory.getAnyInstance()).getSecurityService()
+        .isIntegratedSecurity();
+  }
+
+  void populateMethodAuthorizer(String methodAuthorizerName, Set<String> parameterSet,
+      QueryConfigService queryConfigService) {
+    QueryConfigService.MethodAuthorizer methodAuthorizer =
+        new QueryConfigService.MethodAuthorizer();
+    methodAuthorizer.setClassName(methodAuthorizerName);
+
+    if (parameterSet != null && parameterSet.size() != 0) {
+      List<QueryConfigService.MethodAuthorizer.Parameter> parameters = new ArrayList<>();
+      for (String param : parameterSet) {
```

Comment:
this is super nitpicky and silly but there is param, parameter and parameters.  Maybe change param to value?  Just took a bit for me to see where param was used with parameter is param is a parameter... kinda..

Reply:
Done!